infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/NegotiatingState.java

```diff
@@ -166,15 +177,6 @@ private void processNegotiationResponse(LogReplicationMetadataResponseMsg negoti
 
         log.debug("Process negotiation response {} from {}", negotiationResponse, fsm.getRemoteClusterId());
 
-        /*
-         * If the version are different, report an error.
-         */
-        if (!negotiationResponse.getVersion().equals(metadataManager.getVersion())) {
```

Comment:
we will have to consider a case where the 2 versions are incompatible and the FSM should stop.

Reply:
But will this be part of a plugin? who determines that 2 versions are compatible or not?

Reply:
yes it would be in the plugin.  The plugin determines the compatibility.